nixos/modules/services/home-automation/zwave-js-ui.nix

```
@@ -0,0 +1,86 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
+    mkIf
+    mkEnableOption
+    mkOption
+    types
+    ;
+  cfg = config.services.zwave-js-ui;
+in
+{
+  options.services.zwave-js-ui = {
+    enable = mkEnableOption "zwave-js-ui";
+    serialPort = mkOption {
+      type = types.path;
+      description = ''
+        Serial port for the Z-Wave controller.
+
+        Used for permissions only; must be additionally set in the application
+      '';
+      example = "/dev/ttyUSB0";
+    };
+  };
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !config.services.zwave-js.enable;
+        message = "zwave-js-ui conflicts with zwave-js";
+      }
+    ];
+    systemd.services.zwave-js-ui = {
+      environment = {
+        STORE_DIR = "/var/lib/zwave-js-ui";
```

```suggestion
        STORE_DIR = "$STATE_DIRECTORY";
```

I think you can reference it like that.

---

Unfortunately, systemd doesn't expand variables in environment entries

---

The systemd way would be `%S/zwave-js-ui` where %S will be expanded to `/var/lib`. https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Specifiers